File: Data Structures & Algorithms/max-water-container/submission-7.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0;
        int r = heights.length - 1;
        int maxArea = 0;

        while (l < r) {
            System.out.println(l + "  " + r);
            int minHeight = Math.min(heights[l], heights[r]);
            maxArea = Math.max(maxArea, (r - l) * minHeight);
            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
       return maxArea;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-7.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        max_product = float("-inf")
        curr_max = 1
        curr_min = 1
        

        for num in nums:
            tmp = curr_min * num
            curr_min = min(curr_max * num, curr_min * num, num)
            curr_max = max(curr_max * num, tmp, num)
            max_product = max(max_product, curr_max)
        return max_product
        